### ISSUE TITLE
arm: build error for arm (wily) cross build

### ISSUE THREAD

**Issue by hseok-oh** [dotnet/corefx#10113]
cross build for Ubuntu 15.10 ARM
rootfs option: arm wily
build option: cross arm native debug

```
-- Check for working C compiler: /usr/bin/clang-3.5 -- broken
CMake Error at /usr/share/cmake-2.8/Modules/CMakeTestCCompiler.cmake:61 (message):
  The C compiler "/usr/bin/clang-3.5" is not able to compile a simple test
  program.

  It fails with the following output:

   Change Dir: /home/hseok82/rbp3/corefx/bin/obj/Linux.arm.Debug/Native/CMakeFiles/CMakeTmp



  Run Build Command:/usr/bin/make "cmTryCompileExec1892218345/fast"

  /usr/bin/make -f CMakeFiles/cmTryCompileExec1892218345.dir/build.make
  CMakeFiles/cmTryCompileExec1892218345.dir/build

  make[1]: Entering directory
  `/home/hseok82/rbp3/corefx/bin/obj/Linux.arm.Debug/Native/CMakeFiles/CMakeTmp'


  /usr/bin/cmake -E cmake_progress_report
  /home/hseok82/rbp3/corefx/bin/obj/Linux.arm.Debug/Native/CMakeFiles/CMakeTmp/CMakeFiles
  1

  Building C object
  CMakeFiles/cmTryCompileExec1892218345.dir/testCCompiler.c.o

  /usr/bin/clang-3.5 -target armv7-linux-gnueabihf -mthumb -mfpu=vfpv3
  --sysroot=/home/hseok82/rbp3/corefx/cross/rootfs/arm -o
  CMakeFiles/cmTryCompileExec1892218345.dir/testCCompiler.c.o -c
  /home/hseok82/rbp3/corefx/bin/obj/Linux.arm.Debug/Native/CMakeFiles/CMakeTmp/testCCompiler.c


  Linking C executable cmTryCompileExec1892218345

  /usr/bin/cmake -E cmake_link_script
  CMakeFiles/cmTryCompileExec1892218345.dir/link.txt --verbose=1

  /usr/bin/clang-3.5 -target arm-linux-gnueabihf
  -B/home/hseok82/rbp3/corefx/cross/rootfs/arm/usr/lib/gcc/arm-linux-gnueabihf
  -L/home/hseok82/rbp3/corefx/cross/rootfs/arm/lib/arm-linux-gnueabihf
  --sysroot=/home/hseok82/rbp3/corefx/cross/rootfs/arm
  CMakeFiles/cmTryCompileExec1892218345.dir/testCCompiler.c.o -o
  cmTryCompileExec1892218345 -rdynamic

  /usr/bin/arm-linux-gnueabihf-ld: cannot find crt1.o: No such file or
  directory

  /usr/bin/arm-linux-gnueabihf-ld: cannot find crti.o: No such file or
  directory

  /usr/bin/arm-linux-gnueabihf-ld: cannot find crtbegin.o: No such file or
  directory

  /usr/bin/arm-linux-gnueabihf-ld: cannot find -lgcc

  /usr/bin/arm-linux-gnueabihf-ld: cannot find -lgcc_s

  /usr/bin/arm-linux-gnueabihf-ld: cannot find -lc

  /usr/bin/arm-linux-gnueabihf-ld: cannot find -lgcc

  /usr/bin/arm-linux-gnueabihf-ld: cannot find -lgcc_s

  /usr/bin/arm-linux-gnueabihf-ld: cannot find crtend.o: No such file or
  directory

  /usr/bin/arm-linux-gnueabihf-ld: cannot find crtn.o: No such file or
  directory

  clang: error: linker command failed with exit code 1 (use -v to see
  invocation)

  make[1]: *** [cmTryCompileExec1892218345] Error 1

  make[1]: Leaving directory
  `/home/hseok82/rbp3/corefx/bin/obj/Linux.arm.Debug/Native/CMakeFiles/CMakeTmp'


  make: *** [cmTryCompileExec1892218345/fast] Error 2

```

**Comment by karelz** [comment]
@hseok-oh does it still repro?

**Comment by hseok-oh** [comment]
@karelz Failed with different error message when I follow new guidance: 
https://github.com/dotnet/corefx/blob/master/Documentation/building/cross-building.md

```
Failed to generate native component build project!
CMake Error at configure.cmake:425 (message):
  Cannot find inotify functions on a Linux platform.
Call Stack (most recent call first):
  CMakeLists.txt:227 (include)
```

**Comment by ianhays** [comment]
I'm not familiar with the arm linux build. 

@mkborg @jkotas @prajwal-aithal may have better context here.

**Comment by mellinoe** [comment]
@leemgs Do you have any more information about this? I'm not familiar with the ARM cross-build work.

**Comment by leemgs** [comment]
@mellinoe, From my experience, the issue results from the below cases. 

First, If theare were not developer package like libc-devel (e.g., gcc-multilib or glibc-devel or libc-devel)

Second, when a developemtn environment was too complex (or too was mixed) with "export LIBRARY_PATH . . . "

In case of another case, I recommend that you will try to find a reason with strace command, 
target$> strace -v -o log -f -e trace=open,fork,execve  ...

**Comment by hseok-oh** [comment]
fixed by #14618
